### PR TITLE
starktech files

### PR DESCRIPTION
**Pull request by loganhart420**

### CODE REVIEW

**Comment by erogol** [comment]
What was that actually?  Looks interesting.

**Comment by loganhart420** [comment]
i’m working on a higher level api to do rapid training on multiple model types like for example making two fuctions, one for normal tacotron2 and one using ddc and all you do is call the function, meant to sync that with my fork as i’m just getting started. if you guys think this might be useful for the whole community i’ll definitely submit a pr once I get it finished.

**Comment by loganhart420** [comment]
I’m also making notebooks explaining the process of training a TTS model and how the models actually work.

**Comment by loganhart420** [comment]
A better word for it is like a cook book with a bunch of predefined recipes where all you need to do is input the parameters that are even useful to change after each experiment like changing the batch size.

**Comment by erogol** [comment]
BTw in case you missed it, there is also a new recipes folder. 

> i’m working on a higher level api to do rapid training on multiple model types like for example making two fuctions, one for normal tacotron2 and one using ddc and all you do is call the function, meant to sync that with my fork as i’m just getting started. if you guys think this might be useful for the whole community i’ll definitely submit a pr once I get it finished.

Do you mean to train them one after another or train them together? 

> I’m also making notebooks explaining the process of training a TTS model and how the models actually work.

Notebooks are definitely useful.

Also, feel free to share your experience as you use the new API. I'm really interested in hearing if there is something stupid, lacking, bad, or whatever. 

Waiting for your PR(s) ... :)

**Comment by loganhart420** [comment]
yea train them one right after another, essentially I’m on a whole new level of lazy and instead of going back to tacotron2 config and disabling ddc or If I wanted to change a dataset I train with instead of going back and changing the audio parameters to fit my dataset i’d rather just call a function where I already defined them. basically just writing all of the training scripts at once and calling them when I want to train something. if that makes any sense.

**Comment by erogol** [comment]
I guess it makes sense when you want to try all the models and pick the best. It is half-way to autoTTS :)

**Comment by loganhart420** [comment]
> I guess it makes sense when you want to try all the models and pick the best. It is half-way to autoTTS :) 

hey sorry for the late reply I wanted to make sure it worked and but yea definitely something like that. 
I'm able to train models with just three lines of code. it is really just to train really quick and only the hyper parameters are changeable. You just give it the data path, batch size, learning rate and a couple other parameters and it sets up everything for you.

it looks like this

trainer = Tacotron2Trainer(..)

model = trainer.tactron2_single_speaker_model()

model.fit()

**Comment by erogol** [comment]
Looks really interesting. 

How different do you think it is from using the recipes and changing the parameters for your model?

**Comment by loganhart420** [comment]
It essentially is a recipe but more defined, for example a tacotron2 model training on ljspeech is going to have a different config than say a tacotron2 model using GST and training on vctk and libriTTS. so what I have done here is just create recipes for the different models and wrap them into a function. This function also creates everything for the trainer with more defined audio configs because the defaults aren't always the same as what is needed for the data set. I went off the available  per-trained model configs in the repo and changed parameters accordingly to each data set and model type. If someone really wants to mess around with the configuration then this wouldn't be for them, but if someone just wants to see how a model using gst compares to a model without it would be as simple as calling the gst model. heres a screenshot below of defining different models, although I've already changed it so you don't have to define a phoneme cache path anymore.
Screenshot from 2021-07-14 09-22-42

**Comment by loganhart420** [comment]
really just removes a lot of having to type actual code out, I'm also trying to create a custom data formatter so all you have to define is a path to wav files and text or meta data files

**Comment by erogol** [comment]
I see.  I think about two things now. 

1. How can we even get rid of the data-related parameters and make it "just run" with no change? 

2. Where should we host "starktech" in 🐸 TTS :) ?

**Comment by loganhart420** [comment]
if you are talking about the data sets everything is defined under the hood here is what my data loader function looks like right as I've been more trying to get the models up and running before I spend more time on making it better.
Screenshot from 2021-07-14 09-55-09
and starktech is actually the name of a project I'm working on, I got access to gpt-3 last year and I was able to recreate how Jarvis from Iron talks, like his dialogue style and combined with my other NLP projects and the massive amount of knowledge that model has I was able to create somewhat of the actual character so now I'm just trying to clone his voice to actually make Jarvis from Iron Man. so we can call this sort of a recipe api, and I'm not really sure what you mean by host, sorry I'm kind of a moron sometimes and if I'm way off on the answer to the data-related params questions let me know I'll try to give a more detailed answer. :)

**Comment by loganhart420** [comment]
Once I have all the code a lot faster and cleaner I'll make a new pr too.
